shotover-proxy/tests/transforms/tee.rs

```diff
@@ -0,0 +1,142 @@
+#[serial]
+async fn test_ignore_matches() {
+    let shotover_manager =
+        ShotoverManager::from_topology_file("tests/test-topologies/tee/ignore.yaml");
+
+    let mut connection = shotover_manager.redis_connection_async(6379).await;
+
+    let result = redis::cmd("SET")
+        .arg("key")
+        .arg("myvalue")
+        .query_async::<_, String>(&mut connection)
+        .await
+        .unwrap();
+
+    assert_eq!("42", result);
+}
+
+#[tokio::test(flavor = "multi_thread")]
+#[serial]
+async fn test_ignore_with_mismatch() {
+    let shotover_manager =
+        ShotoverManager::from_topology_file("tests/test-topologies/tee/ignore_with_mismatch.yaml");
+
+    let mut connection = shotover_manager.redis_connection_async(6379).await;
+
+    let result = redis::cmd("SET")
+        .arg("key")
+        .arg("myvalue")
+        .query_async::<_, String>(&mut connection)
+        .await
+        .unwrap();
+
+    assert_eq!("42", result);
+}
+
+#[tokio::test(flavor = "multi_thread")]
+#[serial]
+async fn test_fail_matches() {
+    let shotover_manager =
+        ShotoverManager::from_topology_file("tests/test-topologies/tee/fail.yaml");
+
+    let mut connection = shotover_manager.redis_connection_async(6379).await;
+
+    let result = redis::cmd("SET")
+        .arg("key")
+        .arg("myvalue")
+        .query_async::<_, String>(&mut connection)
+        .await
+        .unwrap();
+
+    assert_eq!("42", result);
+}
+
+#[tokio::test(flavor = "multi_thread")]
+#[serial]
+async fn test_fail_with_mismatch() {
+    let shotover_manager =
+        ShotoverManager::from_topology_file("tests/test-topologies/tee/fail_with_mismatch.yaml");
+
+    let mut connection = shotover_manager.redis_connection_async(6379).await;
+
+    let err = redis::cmd("SET")
+        .arg("key")
+        .arg("myvalue")
+        .query_async::<_, String>(&mut connection)
+        .await
+        .unwrap_err()
+        .to_string();
+
+    let expected = "An error was signalled by the server: The responses from the Tee subchain and down-chain did not match and behavior is set to fail on mismatch";
+    assert_eq!(expected, err);
+}
+
+#[tokio::test(flavor = "multi_thread")]
+#[serial]
+async fn test_subchain_matches() {
+    let _compose = DockerCompose::new("examples/redis-passthrough/docker-compose.yml")
+        .wait_for("Ready to accept connections");
+
+    let shotover_manager =
+        ShotoverManager::from_topology_file("tests/test-topologies/tee/subchain.yaml");
+
+    let mut connection = shotover_manager.redis_connection_async(6379).await;
+    let mut mismatch_chain_redis = shotover_manager.redis_connection_async(1111).await;
+    redis::cmd("SET")
+        .arg("key")
+        .arg("myvalue")
+        .query_async::<_, String>(&mut mismatch_chain_redis)
+        .await
+        .unwrap();
+
+    redis::cmd("SET")
+        .arg("key")
+        .arg("notmyvalue")
+        .query_async::<_, ()>(&mut connection)
+        .await
+        .unwrap();
```

Comment:
We should assert that this returns 42 (and do the same for test_subchain_with_mismatch)